src/React/Basic/DOM.purs

```diff
@@ -7,7 +7,7 @@
 
 module React.Basic.DOM where
 
-import React.Basic.DOM.Internal (createElement, createElementNoChildren)
+import React.Basic (createElement)
```

Comment:
This reverses the dependencies which means we can't reexport things. I'd really like to be able to use a single qualified import of `React.Basic` which is why I went with the internal module.

Reply:
Also, I think it would be good to give a name to `unsafeCoerce :: String -> ReactComponent` and then to use that, since that is actually a _safe_ coercion if you consider `ReactComponent` to mean "something I pass to `createElement`".

Reply:
Good point, I'll add that function.

As for module dependencies, the convenience of reexporting locks this library into DOM-only React, when in reality the only the only DOM-specific code in this library is the DOM module.  This structure allows alternate base components, like native (and native-web).

If the `div` component was actually defined as a component instead of React having special support for strings-as-components, they would be imported from `react-dom`, not `react`.  We might also want to add `purescript-react-basic-dom` at some point with a `React.Basic.DOM.Render` module.  This DOM module could live in either one since it doesn't actually depend on the `react-dom` library.